src/actions/layers/Sources.ts

```diff
@@ -0,0 +1,23 @@
+import TransformableImage from "../../transformation/TransformableImage";
+
+export class ImageLayer extends TransformableImage {
+  // TODO So an ImageLayer needs to stringify itself differently
+  // instead of joining with '/' we join with ','
+  // wil be reused probably
+  toString(): string {
```

Comment:
This is a bit trickier than that. A layer action - and all it's derivatives (cutter, style_transfer, etc.) - is built using more than one component, unlike all other actions (component = anything between slashes in the transformation).
We should probably schedule a short session detailing this, however it'll end up a lot more complicated than this join. 

Another issue with our complicated type-hierarchy is that a lot of the layer behaviour is shraed among all layer types, however they can't have a shared super-class - since each one has a super class already (e.g. this class has `TransformableImage`, and the video layer will have another). That's why in PHP there are traits everywhere, and that's why in Kotlin it's just completely different altogether.